### PR TITLE
Make sure the Thread is always displayed when it's open

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1606

### CODE REVIEW

**Review by FabianDevel — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 1]
```diff
@@ -175,6 +176,20 @@ open class TwoPaneFragment : Fragment() {
...
+        ensureThreadIsOpen()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        ensureThreadIsOpen()
+    }
+
+    private fun ensureThreadIsOpen() {
+        if (mainViewModel.isInThread) openThread(mainViewModel.currentThreadUid.value!!)
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:190]:
```suggestion
        mainViewModel.currentThreadUid.value?.let { currentThreadUid ->
           if (mainViewModel.isInThread) openThread(currentThreadUid)
        }
```

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:190]: The whole purpose of `isInThread ` is to avoid writing that :D

Comment (tevincent) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:180]: Yes but `!!` are evil. Don't be so close to the dark side of the force.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:1]: Why is this needed? Because it looks a bit heavy to call `openThread()` again, this often

**Review by LunarX — CHANGES_REQUESTED** [review]
